### DISCUSSION TITLE
如何进行手动模式？

### DISCUSSION THREAD

**Discussion by 1zhouzhou** [Yuukiy/JavSP#152] [问答 | Q&A] [answered]
根据wiki里的介绍，尝试启动手动模式，都没有意外的失败了，不知道作者大大能不能把步骤方法写得更详细一些，比如“参数-c”具体怎么使用。感谢作者大大无私分享。

**Comment by 1zhouzhou** [comment, marked as answer]
刚刚不死心又尝试了几次，居然成功了，写一个更详细的步骤，希望能采用放入实例里：
![1212233](https://github.com/Yuukiy/JavSP/assets/13687151/15ecb8e9-9523-4077-b650-fd014b13d1f6)
如图所示：“win键+R” 输入cmd启动CMD.exe，把你的JavSP.exe的路径贴上去，比如我的是【D:\JavSP-v1.5.1-Windows-amd64】，然后输入“空格-c”，重点来了，你复制修改新建出来的manual.ini配置文件的路径也要写清楚，否则会运行失败，如图所示去输入，然后“空格-m all”；至此全部的命令行输入就完成了，回车运行即可。

**Reply by Yuukiy** [reply]
![image](https://github.com/Yuukiy/JavSP/assets/76897913/9c57d3b2-39c8-4a58-8ea1-4a357a4ef7b2)
以命令行模式运行的程序，无论Windows还是Linux，当前目录的概念都很重要。如图红框是你的当前目录，由于这个目录既不是JavSP.exe所在的目录，也不是你的ini文件所在的目录，所以你的JavSP.exe和ini文件都需要使用完整的路径。

假设你的当前目录是JavSP.exe所在的文件夹，那exe和ini都只需要输入文件夹名称，不需要带上完整路径也可以的。例如假设你的当前目录是`D:\JavSP-v1.5.1-Windows-amd64`，那只需要输入`javsp -c manual.ini -m all`（Windows的文件系统默认不区分大小写，全小写也OK的。exe文件本身的`.exe`后缀也可以省略，所以只输`javsp`也可以），效果类似这样
```
D:\JavSP-v1.5.1-Windows-amd64>javsp -c manual.ini -m all
==================================
    Jav Scraper Package: v1.5.1
==================================
```
要在JavSP.exe所在的目录启动命令行窗口的话，在相应文件夹下Windows的资源管理器地址栏输入`cmd`然后回车就可以
![GIF 2023-12-02 20-26-24](https://github.com/Yuukiy/JavSP/assets/76897913/7ff8ff17-ccd4-4673-9c23-97d45bc48a0a)

**Reply by 1zhouzhou** [reply]
谢谢作者传授，真的学到了。
这里其实我有另一个疑问，想跟作者探讨一下。在我看来，手动模式应对重复番号的应用场景，好像是其唯一的应用，既然如此，把文件名改成网站的番号，比如，ABC-001.mp4 → ABC-001-12-12-12.mp4，使用手动模式时，是否直接提取文件名刮削即可？如此这般，手动模式也可以改名为“视频文件名刮削模式”？

**Reply by Yuukiy** [reply]
手动模式的初衷是应对番号识别不对的场景，用来人工指定番号。重复番号也可以视为一种特殊的“番号识别不对”。不采用直接提取文件名的方式有一些初衷：1. 改名并不比直接指定番号来得容易，因为一次整理中识别失败的影片可能来自多个文件夹；2. 手动指定番号时用户对番号的校正可能能够用来帮助改进番号识别，只不过这个功能需要搭一个服务器收集校正信息，待开发功能很多，我就一直鸽着了。

**Reply by 1zhouzhou** [reply]
早一些时间，我用手动模式用来解决了中杠后带字母的一些番号的刮削，大概20部+，比如MKD-S03；非常成功。这个玩法，作者或许可以写入wiki里。
